program/src/state.rs

```diff
@@ -3,7 +3,8 @@ use std::str::FromStr;
 use borsh::{BorshDeserialize, BorshSerialize};
```

Comment:
I am not able to comment on `MigrationState::from_account_info` since its not changed in this diff.  But on line 31 you have `let ua = Self::deserialize(&mut data.as_ref())`.  This allows for data to be larger than the expected data size.  You might consider using `try_from_slice()` which would fail if the data was too long (`error: "Not all bytes read"`).

Also I don't see a nonzero discriminator `Key`, so without that one could deserialize a Vec of zeros that was anything greater than `size_of::<MigrationState>()`.  I think in some other versions of `from_account_info` we have check both discriminator and data size which could solve this?  Also just an owner check and PDA check might solve it because then you know it can't be just zeros made by someone else.

Reply:
I want to ultimately have some proper discriminators on Token Metadata accounts, but don't think it's worth implementing that for this contract. With the `is_zeroed` check I added we shouldn't have to worry about data sizes that are to long being deserialized.

Reply:
yes if its not zeroed and its owned by the program, then we know it 1) can't be zeros made by someone else (that deserialize into something) or 2) nonzero data made by someone else.